src/main/java/Hari.java

```diff
@@ -0,0 +1,535 @@
+        private String timerEnd; // End time for Event tasks
+        private String deadlineStat; // Deadline for Deadline tasks
+
+        // Constructor for To Do tasks
+        public Task(String summary) {
+            this.summary = summary;
+            this.completion = false;
+            this.taskType = "T";
+            this.deadline = null;
+            this.timerStart = null;
+            this.timerEnd = null;
+        }
+
+        // Constructor for Deadline tasks
+        public Task(String summary, String deadlineStat) {
+            this.summary = summary;
+            this.completion = false;
+            this.taskType = "D";
+            this.deadline = parseDeadline(deadlineStat);
+            this.timerStart = null;
+            this.timerEnd = null;
+        }
+
+        // Constructor for Event tasks
+        public Task(String summary, String startTime, String endTime) {
+            this.summary = summary;
+            this.completion = false;
+            this.taskType = "E";
+            this.deadline = null;
+            this.timerStart = startTime;
+            this.timerEnd = endTime;
+        }
+
+        // Additional method to parse deadline string into LocalDate
+        private LocalDate parseDeadline(String deadlineStat) {
+            try {
+                // Parse the date string into a LocalDate object
+                return LocalDate.parse(deadlineStat, DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+            } catch (Exception e) {
+                // Handle parsing exception (invalid date format)
+                System.out.println("Error when parsing deadline. Please use the format yyyy-MM-dd.");
+                return null;
+            }
+        }
+
+        // To mark as completed
+        public void completionMark() {
+            this.completion = true;
+        }
+
+        // To unmark completed tasks
+        public void incompletionMark() {
+            this.completion = false;
+        }
+
+        // For displaying the X or [ ] depending on completion status
+        public String completionStatus() {
+            return (completion ? "X" : " "); // To display the X or [ ]
+        }
+
+        // For displaying task description
+        public String summaryStatus() {
+            return summary;
+        }
+
+        // For displaying the task type
+        public String taskStatus() {
+            return taskType;
+        }
+
+        // For displaying start time of event tasks
+        public String timerStartStatus() {
+            return timerStart;
+        }
+
+        // For displaying end time of event tasks
+        public String timerEndStatus() {
+            return timerEnd;
+        }
+
+        // For displaying deadline of deadline tasks
+        public String deadlineStatus() {
+            return deadlineStat;
+        }
+
+        // Additional method to check if task description is empty
+        public boolean summaryEmpty() {
```

Comment:
For booleans, I think it should be named to sound like booleans. So in this case, maybe 'isDescriptionEmpty' could be one way to name this.